src/extra/recipes/base.py

```diff
@@ -0,0 +1,75 @@
+
+import h5py
+from dataclasses import dataclass, asdict, is_dataclass
+
+def save_dict(h5grp, obj: Dict[str, Any]):
```

Comment:
What do you think about saving the settings in an xarray Dataset? That allows for storing quite rich arrays with coordinates/attributes, and we wouldn't have to write any code to save/load it from an HDF5 file.

Reply:
We actually discussed it and I tried doing it right now, but the issue is that I have elements inside of the classes that are dctionaries in `CookieboxCalib`, for example. I would need somehow to flatten the structure of those dicts to be able to use xarray.